plugins/api-docs/src/components/ApiExplorerTable/ApiExplorerTable.test.tsx

```diff
@@ -78,4 +80,60 @@ describe('ApiCatalogTable component', () => {
     expect(rendered.getByText(/api2/)).toBeInTheDocument();
     expect(rendered.getByText(/api3/)).toBeInTheDocument();
   });
+
+  it('should display the specified table columns', async () => {
+    const entitiesWithCustomMetadata: Entity[] = entities.map((e, i) => {
+      const { metadata } = e;
+      const customMetadata = {
+        ...metadata,
+        foo: `foo${i}`,
+        bar: ['baz', 'qux', 'quux'],
+      };
+      return { ...e, metadata: customMetadata };
+    });
+    const fooColumn = { title: 'Foo', field: 'entity.metadata.foo' };
+    const barColumn = {
+      title: 'Bar',
+      field: 'entity.metadata.bar',
+      render: ({ entity }: EntityRow) => (
+        <>
+          {entity.metadata.bar &&
+            (entity.metadata.bar as string[]).map(t => (
+              <Chip key={t} label={t} size="small" variant="outlined" />
+          ))}
+        </>
+      ),
+    };
+    const barFilter = { column: 'Bar', type: 'select' } as TableFilter;
+    const columns = [
+      'Name',
+      'Description',
+      'Owner',
+      'Type',
+      fooColumn,
+      barColumn,
+    ];
+    const filters = [
+      'Type',
+      'Lifecycle',
+      barFilter,
+    ] as CustomizableTableProps['filters'];
+
+    const rendered = render(
+      wrapInTestApp(
+        <ApiProvider apis={apiRegistry}>
+          <ApiExplorerTable
+            entities={entitiesWithCustomMetadata}
+            loading={false}
+            columns={columns}
+            filters={filters}
+          />
+        </ApiProvider>,
+      ),
+    );
+
+    expect(rendered.getByText('Name')).toBeInTheDocument();
+    expect(rendered.getByText('Foo')).toBeInTheDocument();
+    expect(rendered.getByText('Bar')).toBeInTheDocument();
```

Comment:
I meant to also verify that any custom filters were visible after clicking on the filters icon. However, I couldn't find a way around the limitation in `jsdom` having a bunch of the `window.location` functionality not implemented.